python_modules/dagster-graphql/dagster_graphql/implementation/asset_checks_loader.py

```diff
@@ -110,13 +106,21 @@ def _fetch_checks(
                     :limit_per_asset
                 ]
 
+        # prefetch the checks and prepare the runs that will be needed to resolve their statuses
+        # note: this would not be necessary if this was async
         all_check_keys = [
             external_check.key
             for external_checks in external_checks_by_asset_key.values()
             for external_check in external_checks
         ]
-        execution_loader = AssetChecksExecutionForLatestMaterializationLoader(
-            self._context.instance, check_keys=all_check_keys
+        check_records = AssetCheckExecutionRecord.blocking_get_many(self._context, all_check_keys)
+        RunRecord.prepare(
```

Comment:
nit: seeing this makes me thing we could use a more explicit name like `prefetch` `prime_cache` or something to be more obvious